deps/parser/parser.jl

```diff
@@ -0,0 +1,203 @@
+)
+
+function julia_function_string(julia_name::String, polymake_name::String, app_name::String, julia_args::String, call_args::String, parameter_string::String )
+    return """
+function $julia_name( $julia_args )
+    $(typestring)application( "$app_name" )
+    return $(typestring)call_function( "$polymake_name$parameter_string", Array{Any,1}([ $call_args ]) )
+end
+
+"""
+end
+
+function julia_method_string(julia_name::String, polymake_name::String, app_name::String, julia_args::String, call_args::String, parameter_string::String )
+    return """
+function $julia_name( $julia_args )
+    $(typestring)application( "$app_name" )
+    return $(typestring)call_method( "$polymake_name$parameter_string", dispatch_obj, Array{Any,1}([ $call_args ]) )
+end
+
+"""
+end
+
+function create_argument_string_from_type( type_string::String, number::Int64 )
+    ## Special case for BigObject (we do not care 'bout the type here)
+    if startswith(type_string,"BigObject")
+        return "arg" * string(number) * "::" * type_translate_list["BigObject"]
+    end
+    ## We only translate the known cases
+    if !haskey(type_translate_list,type_string)
+        throw(UnknownPolymakeType(type_string))
+    end
+    return "arg" * string(number) * "::" * type_translate_list[type_string]
+end
+
+function parse_definition(method_dict::Dict, app_name::String)
+    name = method_dict["name"]
+    arguments = method_dict["args"]
+    mandatory = method_dict["mandatory"]
+    type_params = method_dict["type_params"]
+    is_method = haskey(method_dict,"method_of")
+    ## for now, we use the same name for Julia and Polymake
+    julia_name = name
+    polymake_name = name
+
+    ## Check for option set
+    has_option_set = last(arguments) == "OptionSet"
+
+    ## Make type parameters
+    param_list_header = Array{String,1}()
+    param_list = Array{String,1}()
+    for i in 1:type_params
+        push!(param_list_header,"param$i::String")
+        push!(param_list,"\$param$i")
+    end
+    if length(param_list) >= 1
+        parameter_string = "<" * join(param_list,",") * ">"
+    else
+        parameter_string = ""
+    end
+
+    ## Compute the argument range
+    max_argument_number = length(arguments) - (has_option_set ? 1 : 0)
+    min_argument_number = mandatory
+
+    ## Option set parameter
+    option_set_argument = "option_set::" * type_translate_list["OptionSet"]
+    option_set_parameter = "option_set"
+
+    ## Real arguments
+    if is_method
+        if ! startswith(method_dict["method_of"],"BigObject")
+            throw(UnparsablePolymakeFunction(name))
+        end
+        argument_list_header = Array{String,1}(["dispatch_obj::$(typestring)pm_perl_Object"])
+        ## Fixme: good?
+        min_argument_number = min_argument_number + 1
+        max_argument_number = max_argument_number + 1
+    else
+        argument_list_header = Array{String,1}()
+    end
+    argument_list = Array{String,1}()
+
+    for i in 1:max_argument_number
```

Comment:
I'd be best to make it into a `function argument_list(arguments, max_argument_number)` which
`return argument_list_header, argument_list`